### DISCUSSION TITLE
Freetz-NG Wireguard-Paket - funktioniert das in der 7.57er Firmware? Wenn ja, wie?

### DISCUSSION THREAD

**Discussion by Bodenseematze** [Freetz-NG/freetz-ng#948] [Allgemein] [answered]
Hallo,

ich habe eine FritzBox 7590 mit Freetz-NG - in der Vergangenheit immer noch auf der AVM-Firmware v7.29 aufsetzend.
Dort hatte ich schon seit gefühlten Ewigkeiten das Wireguard-Paket verwendet, um die Box als VPN-Server fungieren zu lassen und mich von außen verbinden zu können.
Das hat immer wunderbar funktioniert!

Jetzt habe ich mir ein neues Image basierend auf der aktuellen v7.57 Firmware von AVM gebaut - seit v7.50 hat AVM ja selber ein Wireguard integriert...
Das habe ich mir mal angeschaut und für mich entschieden, dass ich das zu wenig Einflussmöglichkeit auf die Server-Parameter habe und wollte weiterhin das in Freetz integrierte Paket nutzen.

Nur leider bekomme ich mit dem auf v7.57 basierenden Image keine Verbindung hin bzw. der Wireguard-Client baut zuerst eine Verbindung auf und wartet dann auf das Handshake vom Server und das kommt wohl nicht...
Ich habe das 7.57er Image sowohl mit AVM-Wireguard gebaut als auch mit Removal-Patch - da gibt es keinen Unterschied!

In der zweiten Partition meiner Box habe ich noch das auf v7.29 basierende Image - wenn ich das boote, klappt die Verbindung mit dem Wireguard-Server wieder ohne Probleme.

Was hat sich also zw. 7.29 und 7.50/7.57 geändert bzw. gibt es jemand, bei dem der Wireguard-Server als Freetz-NG-Paket in eine aktuellen 7.57 funktioniert?
Gibt es irgendwelche Device-Schnittstellen, die anders heißen bzw. nicht mehr funktionieren?

Oder: muss ich etwas an meiner Konfiguration ändern, damit es wieder funktioniert?

**Comment by aus-sua** [comment]
Nun ja...
Ich denke, mit so einer Blümchen-Prosa-"Fehlerbeschreibung" wird Dir keiner hier helfen können/wollen.
Es fehlen logs, WG-konfigs und noch nicht einmal die reale Fehlermeldung des/der WG-Server/-Clients hast Du
mitgeliefert.
Unabhängig davon, hattest Du nicht Deine busybox komplett umgemodelt/erweitert, da würde ich wohl erst
einmal vergleichen...

**Reply by Bodenseematze** [reply]
Das hier ist das Wireguard-Protokoll auf dem (Windows-)Client, wenn es schiefgeht (1.2.3.4 ist meine öffentliche IPv4, :5 ist der verwendete Server-Port des Wireguard Servers):
`2024-02-06 08:36:00.455: [TUN] [Zuhause] Using WireGuardNT/0.10`
`2024-02-06 08:36:00.455: [TUN] [Zuhause] Enabling firewall rules`
`2024-02-06 08:36:00.434: [TUN] [Zuhause] Interface created`
`2024-02-06 08:36:00.457: [TUN] [Zuhause] Dropping privileges`
`2024-02-06 08:36:00.457: [TUN] [Zuhause] Setting interface configuration`
`2024-02-06 08:36:00.458: [TUN] [Zuhause] Peer 1 created`
`2024-02-06 08:36:00.462: [TUN] [Zuhause] Sending keepalive packet to peer 1 (1.2.3.4:5)`
`2024-02-06 08:36:00.462: [TUN] [Zuhause] Sending handshake initiation to peer 1 (1.2.3.4:5)`
`2024-02-06 08:36:00.462: [TUN] [Zuhause] Monitoring MTU of default v6 routes`
`2024-02-06 08:36:00.462: [TUN] [Zuhause] Interface up`
`2024-02-06 08:36:00.463: [TUN] [Zuhause] Setting device v6 addresses`
`2024-02-06 08:36:00.468: [TUN] [Zuhause] Monitoring MTU of default v4 routes`
`2024-02-06 08:36:00.468: [TUN] [Zuhause] Setting device v4 addresses`
`2024-02-06 08:36:00.470: [TUN] [Zuhause] Startup complete`
`2024-02-06 08:36:05.523: [TUN] [Zuhause] Sending handshake initiation to peer 1 (1.2.3.4:5)`
`2024-02-06 08:36:10.674: [TUN] [Zuhause] Handshake for peer 1 (1.2.3.4:5) did not complete after 5 seconds, retrying (try 2)`

Meine Server-Wireguard-Konfigurationsdatei im Freetz-NG-Paket sieht folgendermaßen aus (etwas angepasst, um private Infos zu "verstecken"):
`[Interface]`
`PrivateKey  = **************************************************`
`ListenPort  = 55555`
`FwMark      = 03`
` `
`[Peer]`
`PublicKey   = *********************************************`
`AllowedIPs  = 10.168.1.120/32`

Als IPv4 Server-Adresse habe ich unter Einstellungen *10.168.1.10/24* angegeben, IPv6 ist leer (verwende ich nicht); als Netzwerkschnittstelle ist dort *wg0* angegeben  - die restlichen Eingabefelder sind leer...

bzgl. Logging: da ich ja "entfernt" bin kann ich zum Fehlerzeitpunkt nicht auf's Freetz-Logging zugreifen - wenn ich aber später wieder heim gekommen bin, konnte ich nichts passendes dazu im Freetz-Log finden...

> Unabhängig davon, hattest Du nicht Deine busybox komplett umgemodelt/erweitert, da würde ich wohl erst
einmal vergleichen...

Was meinst Du damit?

**Comment by fda77** [comment]
Dein Log zeigt dass du dich irgend wo hin verbinden willst. Ob und was da läuft sieht man nicht. Daher wäre die Logs (zu dem Moment des Connects) vom Server aufschlussreicher

**Comment by fda77** [comment]
Zusammenfassung:
 - irgend was geht irgend wie nicht
 - 0 Erkenntnis
 - hat mich nur sinnlos Zeit gekostet

**Comment by Bodenseematze** [comment]
ich habe z.Zt. wenig Zeit deswegen habe ich es einfach mal laufen lassen wie es ist.
Per Wireguard und der 7.57er Firmware komme ich immer noch nicht rein.

Auf dem Server steht bzgl. dem Wireguard Service überhaupt nichts im Logging - also vermute ich aktuell, dass ich keinen Zugriff mehr von außen auf den Wireguard-Port erhalte.
Da (vermutlich durch das laufende hin- und her-booten von 7.29 und 7.57) aktuell auch mein ssh-Login nicht mehr funktioniert, kann ich auf der Box auch nur schwer nachschauen, was bzgl. Portfreigaben gesetzt ist...

Ich werde auf jeden Fall hier berichten (und sei es nur für jemand anderes, der ähnliche Probleme hat), wenn ich wieder etwas mehr Zeit gefunden habe und das Problem weiter analysiert habe.


Trotz allem würde mich schon interessieren, ob es Leute gibt, die das Wireguard-Paket aus Freetz-NG im Zusammenspiel mit 7.5x-Firmware (erfolgreich) verwenden!

**Reply by Bodenseematze** [reply]
So, ich hatte jetzt etwas Zeit zu forschen und mein ssh-Login funktioniert wieder ;-)

Also habe ich mich mal ein wenig auf der laufenden Box umgesehen und gemerkt, dass ich etwas vergessen hatte - ich hatte seit ich Wireguard als Freetz-Paket verwende, folgendes in der ar7.cfg eingetragen:
```
`    voip_forwardrules = ""udp 0.0.0.0:5060 0.0.0.0:5060 # SIP",`
`                            "tcp 0.0.0.0:5060 0.0.0.0:5060 # SIP",`
`                            "tcp 0.0.0.0:<myhttpsport> 0.0.0.0:5443 0 mark 1 # HTTPS",`
`                             "udp 0.0.0.0:7078+20 0.0.0.0:7078 # RTP",`
`                            "udp 0.0.0.0:<mywgport> 0.0.0.0:<mywgport> 0 mark 3 # wireguard";`
```

<myhttpsport> ist der Webserver-Port, den ich von aussen verwende,
<mywgport> ist der o.a. WireGuard-Port, den ich in der Config eingetragen habe.

Auf der Firmware 7.29 kann ich in der AVM-Weboberfläche unter "Diagnose/Sicherheit" auch die geöffneten Ports sehen (da steht dann zwar überall als FritzBox-Dienst "Telefonie (RTP)" - aber das ist ja unerheblich).
Auf der Firmware 7.57 steht dort nur noch der SIP-Port 5060 und der HTTPS-Port - der kommt aber vermutlich aus der Einstellung "Internetzugriff auf die FRITZ!Box über HTTPS aktiviert" (Internet/Freigaben) und ist in der ar7.cfg wohl hier zu finden (?):
```
`websrv {`
`    port = "80";`
`    https_port = "<myhttpsport>";`
`    wanaccess = 1;`
`    read_timeout = 15m;`
`    request_timeout = 30s;`
`    keepalive_timeout = 5m;`
`    nokeepalive = "*";`
`    errordir = "/usr/www/html/errors";`
`    webdir = "/usr/www";`
`    cgidir = "cgi-bin";`
`    indexfn = "index.var", "index.htm", "index.html";`
`    users_only_for_https = no;`
`    cors_allow_origins = "http*://scope.avm.de";`
`    cors_allow_headers = "SOAPACTION", "Content-Type", "Origin";`
`    cors_allow_methods = "GET", "POST", "OPTIONS";`
`    cors_max_age = 1d;`
`}`
```

Also scheinen die Angaben und voip_forwardrules in Firmware 7.57 (7.5x?) nicht mehr zu funktionieren...

Jetzt ist natürlich die Frage, was denn stattdessen funktioniert?!

Wenn ich das Script vom Package avm-portfw richtig interpretiere, setzt dieses den Eintrag "internet_forwardrules" in der ar7.cfg
Den Eintrag gibt es bei mir überhaupt nicht!?

Bei mir gibt es Einträge für voip_forwardrules, voip_ip6_forwardrules, tr069_forwardrules, tr069_ip6_forwardrules und das war's...

Achja, auch unter Firmware 7.29 finde ich keinerlei Logging im Syslog vom Wireguard - nur beim Aufruf von "wg showinfo" auf der Kommandozeile (oder eben in der Freetz-GUI unter "Status/wireguard" - das scheint mir die Ausgabe von "wg showinfo" zu sein)
EDIT: soweit ich im Internet recherchiert habe, müsste man vom wireguard Kernel-Modul das Logging einschalten, um mehr zu sehen.
Es heißt dann, dass man "echo module wireguard +p > /sys/kernel/debug/dynamic_debug/control" machen soll - dass dazu aber der Kernel "dynamisches Debugging" unterstützen muss - ich denke nicht, dass das unser AVM-/Freetz-Kernel kann, oder?
Oder man soll ein "options wireguard dyndbg=+p"  in "/etc/modprobe.d/wireguard.conf" schreiben.
Wie kann ich das auf der Box machen? "/etc/modprobe.d" ist doch nicht beschreibbar (und auch nicht nach /var/tmp umgelenkt o.ä.)

**Comment by aus-sua** [comment]
Erst nach mehr als zwei Wochen merkst Du z.B., daß Deine Box nach außen
keine Ports geöffnet hat?!
Und Du hattest das vor zwei Wochen dann nicht einmal kurz getestet
bzw. auch nur nachgesehen?

Junge, junge...
ich sage es 'mal sehr freundlich:
Bei Dir klaffen aber Anspruch und Wirklichkeit extrem weit auseinander!
:-(

Vorschlag zur Güte:
Sichere Deine Einstellungen des FOS und des freetz-NG - und dann:
Setzte Deine 7590er bitte doch erst einmal auf Werkseinstellung zurück und dann
genau so die Einstellungen für das freetz-NG auf "Werkseinstellung".

Stelle damit bitte erst einmal "saubere Verhältnisse" her!

Dann spare Dir bitte alle Änderungen in irgend welchen internen Dateien auf der Box,
spiele bitte auch keine (!) Deiner alten über ca. 10-jährigen gesammelten "Sicherungen" zurück,
sondern richte erst einmal den reinen FOS-AVM-Teil sauber nach den Vorgaben von AVM über
die FOS-Oberfläche ein.

Dort richtest Du auch die Einstellungen für das AVM-wireguard ein, denn lt. Deiner o.g. WG-config
kann ich nichts von Deiner Ansage:
>  ... und für mich entschieden, dass ich das zu wenig Einflussmöglichkeit auf die Server-Parameter habe

erkennen, was das normale AVM-Wireguard nicht auch erst einmal kann und macht!

Wenn Du Dich an die Vorgaben/Einrichtungen von AVM hältst,
dann sollte es mich da doch sehr wundern, wenn dann die AVM-WG-Verbindungen
nicht normal funktionieren!

Dann sehen wir weiter...

**Reply by Bodenseematze** [reply]
> Erst nach mehr als zwei Wochen merkst Du z.B., daß Deine Box nach außen keine Ports geöffnet hat?!
> Und Du hattest das vor zwei Wochen dann nicht einmal kurz getestet bzw. auch nur nachgesehen?

Ich war so fixiert auf das Server-Logging, dass ich tatsächlich nicht dran gedacht hatte, einen externen Portscan zu machen.
Ich habe ja die meiste Zeit die Firmwareversion 7.29 gebootet und nur selten die 7.57 (um dann von der Arbeit aus kurz einen Verbindungstest zu machen - zu viel mehr war nicht Zeit)...
...dass man die offenen Ports auch bei der AVM WebGUI sehen kann, habe ich auch nicht auf dem Schirm gehabt.
Also diesbezüglich: Asche auf mein Haupt!

> Dort richtest Du auch die Einstellungen für das AVM-wireguard ein, denn lt. Deiner o.g. WG-config kann ich nichts von Deiner Ansage:
> 
> > ... und für mich entschieden, dass ich das zu wenig Einflussmöglichkeit auf die Server-Parameter habe
> 
> erkennen, was das normale AVM-Wireguard nicht auch erst einmal kann und macht!

Ach ja? Wo kann ich z.B. im AVM-Wireguard IP-Adressbereiche für den Server und für die Peers angeben?
Soweit ich mich an meinen Test erinnern kann, nimmt er sogar für die Peers "irgendwelche" automatischen IP-Adressen aus dem gleichen Adressbereich wie die Box - spannt also noch nicht einmal ein eigenes Netz für die Wireguard-Clients auf....

Also ich bleibe dabei: für mich ist der AVM-WireGuard nichts!

**Comment by Bodenseematze** [comment]
So, ich habe jetzt das gemacht, was auch das Paket AVM-portfw macht - ich habe die Freigabe der Ports unter "internet_forwardrules" eingetragen.

Und siehe da: jetzt geht wieder alles :-)

@fda77: was ich aber jetzt auch definitiv sagen kann: wenn sowohl AVM's Wireguard als auch das Freetz-NG-Paket beim Compilieren freigeschaltet sind, kommen sich die beiden ins Gehege (soweit ich das gesehen habe, gibt es dann zwei "wireguard" Kernel-Module und die greifen teilweise wohl gegenseitig auf die Konfigurationen zu - zumindest hat der AVM wireguard plötzlich den von mir im Freetz-NG konfigurierten Server-Port verwendet und in der Übersicht beim Freetz-Paket war dann auch die AVM-Serverkonfig im Status zu sehen.

Vielleicht liegt das auch daran, weil beide die Schnittstelle wg0 verwenden - man könnte nochmals ausprobieren, ob es dann geht, wenn das Freetz-Paket z.B. wg1 verwendet...

Aber selbst wenn das funktioniert - wer will das?
Ich denke, dass im .config eine entsprechende Abhängigkeit eingebaut werden sollte - wenn das Freetz-NG Wireguard-Paket freigeschaltet wird, sollte der Patch zum Entfernen des AVM-Wireguards fest selektiert werden - zumindest bis um "Expert"-Level;
man verliert allerdings durch den Patch nicht nur Wireguard sondern auch den "normalen" AVM VPN-Client...
(im "Developer"-Level könnte der Patch ggf. auch wieder abgewählt werden...)

**Reply by aus-sua** [reply]
Tja, kaum hält man sich an die Vorgaben und liest ggf. mal die
immer noch gute Hilfe der Doku/make menuconfig
dann geht es auf einmal...
Mir geht das auch immer gegen den Strich!

Wenn ich schon lese:
> ich habe jetzt das gemacht, was auch das Paket AVM-portfw macht -
>  ich habe die Freigabe der Ports unter "internet_forwardrules" eingetragen.

Warum benutzt Du dann nicht einfach AVM-portfw und sparst Dir das händische Edit,
wenn Du 'eh nicht weißt, was/warum Du etwas machst? 
Zumindest bis FOS 07.57 [funktioniert](https://github.com/Freetz-NG/freetz-ng/commit/e7cb0e9e4bffc31d96ae0d0163cf05b69693dff7#comments) es ja noch wie erwartet!

Nur 'mal zum nachdenken:
Ja, wer sowohl AVM-wg als freetz-NG-wg gleichzeitig (!) benutzt, bei dem kommen
die sich gegenseitig wohl in die Quere!
Die Benutzung ist eben endweder den einen ODER den anderen!
Mitdenken!

Und immer schön noch Forderungen stellen...
Damit aber jetzt noch Abhängkeiten in die menuconfig einzubauen, nur weil wieder
einmal einer ggf. zu blöd ist, halte ich für kontraproduktiv!
Dann müßte ich z.B. mit Erstellen des Image ja schon entscheiden, ob ich AVM- oder freetz-NG-wg
benutze und jeder Wechsel bedeutet ein eigenes Image!

Der ganze ist, wie ja so oft, verschwendete Zeit, weil wieder einmal:
- keine Doku gelesen
- nicht mitgedacht
- keine Infos geliefert
- "heimliche"/ungenannte config-Änderungen (aka "ich mache da 'mal irgend etwas")

abgeliefert wurde...

Schön ist, daß es jetzt eben doch so einfach funktioniert, wenn... (s.o.)

**Reply by Bodenseematze** [reply]
> Tja, kaum hält man sich an die Vorgaben und liest ggf. mal die immer noch gute Hilfe der Doku/make menuconfig dann geht es auf einmal... Mir geht das auch immer gegen den Strich!

Welche Vorgaben?

> Wenn ich schon lese:
> 
> > ich habe jetzt das gemacht, was auch das Paket AVM-portfw macht -
> > ich habe die Freigabe der Ports unter "internet_forwardrules" eingetragen.
> 

> Warum benutzt Du dann nicht einfach AVM-portfw und sparst Dir das händische Edit, wenn Du 'eh nicht weißt, was/warum Du etwas machst? Zumindest bis FOS 07.57 [funktioniert](https://github.com/Freetz-NG/freetz-ng/commit/e7cb0e9e4bffc31d96ae0d0163cf05b69693dff7#comments) es ja noch wie erwartet!

Ich weiß nicht, wie es bei Dir aussieht - aber zum Einen wird da in der GUI aus einem Mehrzeiler ein Einzeiler - da passt was mit dem Parsing nicht (wollte ich eigentlich als nächstes meine Hilfe anbieten, um das im portfw zu verbessern - muss mich aber erst mal in die GUI-Generierung von Freetz einarbeiten)...
...und zum Anderen, habe ich Angst, was das Script dann bei "falschen" (für den Parser) Angaben versucht, in die ar7.cfg zu schreiben - da mache ich das lieber direkt - da gibt es dann keinen Umweg, der wieder neue Problem verursacht/verursachen kann.

> Nur 'mal zum nachdenken: Ja, wer sowohl AVM-wg als freetz-NG-wg gleichzeitig (!) benutzt, bei dem kommen die sich gegenseitig wohl in die Quere! Die Benutzung ist eben endweder den einen ODER den anderen! Mitdenken!

Klar - dieser Hinweis ging eigentlich an fda77, weil er diesbezüglich im ShellInABox-Thread danach gefragt hatte - ob ich die Koexistenz ausprobiert hätte oder ob er da eine Abhängigkeit einbauen soll.

> Und immer schön noch Forderungen stellen... Damit aber jetzt noch Abhängkeiten in die menuconfig einzubauen, nur weil wieder einmal einer ggf. zu blöd ist, halte ich für kontraproduktiv! Dann müßte ich z.B. mit Erstellen des Image ja schon entscheiden, ob ich AVM- oder freetz-NG-wg benutze und jeder Wechsel bedeutet ein eigenes Image!

Ich stelle keine Forderungen - ich gebe Antwort auf eine Frage, die in einem anderen Thread gestellt wurden.

Und ja, wenn es mit Koexistenz geht - umso besser; das müsste aber erst einmal noch genauer ausprobiert werden.
Bei meinen Tests (und ja, ich hatte immer nur _entweder_ das eine oder das andere aktiviert - niemals beide gleichzeitig; wobei die Deaktivierung bei AVM ja darin besteht, keine Einträge mehr zu haben?!) ging es nicht - habe aber selber beim Schreiben der Antwort gemerkt, dass es vielleicht doch noch Möglichkeiten gibt (andere Netzwerkschnittstelle)...

> Der ganze ist, wie ja so oft, verschwendete Zeit, weil wieder einmal:
>     * keine Doku gelesen

Ich lese sehr wohl Doku - nur muss ich da an die richtigen Stellen geraten - und das ist nicht so einfach, wenn man nicht genau weiß, woran es liegt / liegen könnte.
Also habe ich versucht, irgendeine Art von Logging zu präsentieren - leider nur auf dem Client weil der Server keine hergegeben hat.

>     * nicht mitgedacht

Ja, den Schuh muss ich mir anziehen - dass es an den Port-Freigaben liegen könnte, hätte ich selber merken müssen!

>     * keine Infos geliefert

s.o. - ich habe die Infos geliefert, die mir zur Verfügung standen.

>     * "heimliche"/ungenannte config-Änderungen (aka "ich mache da 'mal irgend etwas")

Anders herum gefragt: bekommt man das Wireguard-Paket in Freetz-NG auch ohne eigenständig (oder über AVM-portfw) eingepflegte Port-Freigabe zum Laufen?

Wenn ich Zeit habe, schaue ich mal im Wiregurat-Paket nach, ob da was vorgesehen ist, was bei mir nicht funktioniert hat bzw. falls nicht, kann ich versuchen, da was einzubauen...

> Schön ist, daß es jetzt eben doch so einfach funktioniert, wenn... (s.o.)

Genau - eigentlich hätte ich nur wissen müssen, dass 
  a) eine händische Portfreigabe notwendig ist (da habe ich mich nicht mehr dran erinnert, weil es schon so lange her war)
  und 
  b) dass "voip_forwardrules" schon länger nicht mehr unterstützt werden bzw. durch "internet_forwardrules" abgelöst wurden
     (wobei es dann halt auch noch verwirrend war, dass es in 7.29 noch funktioniert hat, obwohl das ja wohl bereits nach 6.8 entfernt wurde)

...aber im Nachhinein ist man immer schlauer!

**Comment by fda77** [comment, marked as answer]
Ihr macht mich wahnising!
§1 Wer selbst an seiem Freetz herumpfuscht sag das und meldet hier nicht ein Problem!

> Also scheinen die Angaben und voip_forwardrules in Firmware 7.57 (7.5x?) nicht mehr zu funktionieren...

Wahrscheinlich ist aus genau diesem Grund das Paket "AVM-rules" nur bis fos 6.8 nutzbar!!!!!!!!!
 https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/avm-rules/Config.in#L6

und https://freetz-ng.github.io/freetz-ng/make/avm-rules.html#deprecated

Diese ganz Scheise geht mir sowas von auf die Eier!
Ich setzte diese Diskussion auf ignorieren, erwartet keine weiten Kommentar von mir

**Reply by Bodenseematze** [reply]
Auch wenn Du das auf "ignorieren" setzt - kann ich das so nicht stehen lassen!

Woher soll ein "normaler" Benutzer wissen, dass
  a) ein Paket "AVM-rules" früher mal dafür da war, voip_forwardrules anzupassen?
  b) es dann nach v6.8 nicht mehr auswählbar war - aber Einträge in voip_forwardrules immer noch funktionierten (eben bis einschl. 7.29)?

Vermutlich war es bei mir tatsächlich so, dass ich damals über AVM-rules die ursprüngliche Freigaben eingetragen hatte und dann als das Wireguard-Paket dazu kam, habe ich vermutlich (weil keine GUI mehr da war), das direkt zu den anderen Angaben in die ar7.cfg hinzugefügt - und das ist halt seit dem drin und hatte auch funktioniert; eben bis jetzt...

Und wie sucht jemand im Internet bei sowas? Ich habe z.B. nach "Wireguard freetz" gesucht - aber immer nur Anleirungen gefunden, die sich auf AVM's WG beziehen.
Oder "fritzbox voip_forwardrules" - da findet man gar nichts relevantes.
Oder "fritzbox ar7 forwardrules" - da findet man einiges - aber da wird dann von einem Eintrag "forwardrules" (ohne "voip_", "internet_" sonst was Präfix) geredet.

Mir geht es gewaltig auf die Eier (um mit Deinen Worten zu reden) - dass man hier immer früher oder später von Dir blöd angemacht wird - über Sachen, die man m.E. eben "normalerweise" nicht so aus dem Stehgreif über die Internas weiß und auch nicht mit normalen Suchbegriffen findet.
Im ShellInABox-Thread dachte ich noch, "das klappt ja gut" und "ist ja echt mal konstruktiv" - aber hier war's dann mal wieder Sch***!

Gut - Du bist der Entwickler, der sich über die immer gleichen Fragen vermutlich genervt fühlt - aber Deine Art mit Leuten umzugehen ist (vorsichtig ausgedrückt) oftmals "zu harsch".
Und ich meine das jetzt nicht nur aus meinen eigenen Threads sondern beobachte das auch bei anderen...

Ich finde es übrigens auch etwas seltsam, dass Du meinen eigenen Beitrag, den ich als Antwort markiert hatte, wieder "entmarkiert" hast um Deinen eigenen "Polterbeitrag" als Antwort zu markieren...

Aber gut - so ist es halt...
